tools/Azure.Mcp.Tools.AppService/src/Commands/BaseAppServiceCommand.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using Azure.Mcp.Core.Commands;
+using Azure.Mcp.Core.Commands.Subscription;
+using Azure.Mcp.Tools.AppService.Options;
+
+namespace Azure.Mcp.Tools.AppService.Commands;
+
+public abstract class BaseAppServiceCommand<
+    [DynamicallyAccessedMembers(TrimAnnotations.CommandAnnotations)] TOptions>
+    : SubscriptionCommand<TOptions>
+    where TOptions : BaseAppServiceOptions, new()
+{
+    protected override void RegisterOptions(Command command)
+    {
+        base.RegisterOptions(command);
+        UseResourceGroup();
+        command.Options.Add(_resourceGroupOption);
+    }
+
+    protected override TOptions BindOptions(ParseResult parseResult)
+    {
+        var options = base.BindOptions(parseResult);
+        options.ResourceGroup = parseResult.GetValueOrDefault(_resourceGroupOption);
```

Comment:
base.BindOPtions does this for you when you use UseResourceGroup, remove this line